### ISSUE TITLE
Tutorials on Colab crash

### ISSUE THREAD

**Issue by Soffia-t** [ourownstory/neural_prophet#1399]
**Prerequisites**

* [x ] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in a new virtualenv with only neuralprophet installed, directly from github:
      ```shell
      git clone <copied link from github>
      cd neural_prophet
      pip install .
      ```
    * Checked the Answered Questions on the Github Disscussion board: https://github.com/ourownstory/neural_prophet/discussions
      If you have the same question but the Answer does not solve your issue, please continue the conversation there.
    * Checked that your issue isn't already filed: https://github.com/ourownstory/neural_prophet/issues
      If you have the same issue but there is a twist to your situation, please add an explanation there.
    * Considered whether your bug might actually be solveable by getting a question answered: 
      * Please [post a package use question](https://github.com/ourownstory/neural_prophet/discussions/categories/q-a-get-help-using-neuralprophet)
      * Please [post a forecasting best practice question](https://github.com/ourownstory/neural_prophet/discussions/categories/q-a-forecasting-best-practices)
      * Please [post an idea or feedback](https://github.com/ourownstory/neural_prophet/discussions/categories/ideas-feedback)

**Describe the bug**
When I try to run the tutorials (I tried several), for example (https://colab.research.google.com/github/ourownstory/neural_prophet/blob/main/docs/source/how-to-guides/application-examples/energy_hospital_load.ipynb) on Colab, they always crash the session (our of RAM message) when it tries to m.fit. What I have tried:
-Running a high ram Colab, same thing
-Running it on planetary computer, same thing
-Running on Noteable.io, same thing

I did not touch any of the code on the Tutorials.

P.S. It was happening on my notebooks using NeuralProphet, I thought it might be an error in my code, that's why I went to the tutorials and the error was happening there as well. 

**To Reproduce**

1. Open notebook https://colab.research.google.com/github/ourownstory/neural_prophet/blob/main/docs/source/how-to-guides/application-examples/energy_hospital_load.ipynb on Colab
2. Run as is
3. It will crash the session when it tries to fit
4. Same thing happens when you use a High RAM machine or a GPU

**Expected behavior**

A fix to this so that we can run NeuralProphet on Colab

**Comment by leoniewgnr** [comment]
Hi @Soffia-t, thanks for raising this? Does it crash because it runs out of memory?

**Comment by Soffia-t** [comment]
That is what it says. That it ran out of memory. 
But I have run the energy tutorial ( https://colab.research.google.com/github/ourownstory/neural_prophet/blob/main/docs/source/how-to-guides/application-examples/energy_hospital_load.ipynb) before ( on a normal Colab instance 4Gb) and it was fine. Now, it crashes even on the Pro+ version, 51 Gb of RAM.

**Comment by leoniewgnr** [comment]
I can confirm the problem, I'm currently working on improving the memory performance on NeuralProphet but didn't really find the problem. Do you have an idea where this could be in the fit function? 
I suspect it to be in the [time_dataset.py](https://github.com/ourownstory/neural_prophet/blob/main/neuralprophet/time_dataset.py), but not super sure.

I already improved the stride function [here](https://github.com/ourownstory/neural_prophet/pull/1400). I'll merge it later and then we can check if it still runs out of memory, but I think so :(

**Comment by leoniewgnr** [comment]
![Figure_1](https://github.com/ourownstory/neural_prophet/assets/42536262/4d98ed8b-42d9-4c61-b873-fa62ebefeba8)
This is the memory consumption of a normal fit, so I think we gradually add memory when loading the data and then for fitting it stays constant, which makes sense. I'll check the dataloading process if we can save memory somewhere.

**Comment by leoniewgnr** [comment]
@Soffia-t Do you have insights on how much RAM is used before it crashes?

**Comment by Soffia-t** [comment]
OK, So, I screen recorded the crash. I hope this will give you more info.
[energy_data_example.ipynb - Colaboratory - Google Chrome 2023-08-16 09-58-16.zip](https://github.com/ourownstory/neural_prophet/files/12356514/energy_data_example.ipynb.-.Colaboratory.-.Google.Chrome.2023-08-16.09-58-16.zip)

**Comment by leoniewgnr** [comment]
@Soffia-t, I made some improvements to memory issues, but it still crashes...
I honestly don't know why. Maybe someone can help on this?

**Comment by leoniewgnr** [comment]
resolved with #1418
